pkg/model/components/context.go

```diff
@@ -120,13 +129,15 @@ func IsBaseURL(kubernetesVersion string) bool {
 
 // Image returns the docker image name for the specified component
 func Image(component string, clusterSpec *kops.ClusterSpec) (string, error) {
+
 	if component == "kube-dns" {
 		// TODO: Once we are shipping different versions, start to use them
-		return "gcr.io/google_containers/kubedns-amd64:1.3", nil
+		return GetGoogleImageRepositoryContainer(clusterSpec, "/kubedns-amd64:1.3"), nil
 	}
 
 	if !IsBaseURL(clusterSpec.KubernetesVersion) {
-		return "gcr.io/google_containers/" + component + ":" + "v" + clusterSpec.KubernetesVersion, nil
+		c := component + ":" + "v" + clusterSpec.KubernetesVersion
```

Comment:
Nit: we could also split this out and have two fields - image name & image version.  But NBD